src/app/youTube/services/youTube.service.ts

```diff
@@ -1,17 +1,66 @@
 import { Injectable } from '@angular/core';
-import * as dataJson from '../../../assets/response.json';
-import { Item } from '../../shared/types';
+import {
+  Item,
+  ResultData,
+  Statistics,
+  StatisticsData,
+  WholeVideoData
+} from '../../shared/types';
 
 @Injectable({ providedIn: 'root' })
 export class YouTubeService {
-  items = (dataJson as any).default;
+  API_KEY = 'AIzaSyB0fgxbiZ__2ZQKwB-Wa7kqEsq5cIVOi4Q';
 
-  getItems(): Item[] {
-    return this.items;
+  getItems(searchString: string): Promise<Item[]> {
+    // get list of videos
+    const urlBase = `https://youtube.googleapis.com/youtube/v3/search?key=${this.API_KEY}&part=snippet&maxResults=5`;
+    const params = new URLSearchParams({ q: searchString });
+    const paramsString = params.toString();
+    let url = searchString ? `${urlBase}&${paramsString}` : urlBase;
+
+    return fetch(url)
```

Comment:
You should use HttpClient from angular for making http requests
https://angular.io/tutorial/tour-of-heroes/toh-pt6